pageserver/src/bin/pageserver.rs

```diff
@@ -488,6 +490,13 @@ fn main() -> Result<()> {
     // as a ref.
     let conf: &'static PageServerConf = Box::leak(Box::new(conf));
 
+    let scenario = FailScenario::setup();
+    if fail::has_failpoints() {
+        std::panic::set_hook(Box::new(|_| {
+            std::process::exit(1);
+        }));
+    }
+
```

Comment:
I don't quite understand the purpose of this code.
`FailScenario::setup()` configures fail points specified in the `FAILPOINTS` environment variable. 
As far as I see it is never used in the tests.

Have you set it manually?

Reply:
I suspect @knizhnik wanted to make it testable/usable in manual mode while developing this feature. Either way, I'd leave it be since it's useful.

Reply:
The purpose of this code is to cause pageserver termination in case of panic. `failpoint` module doesn't provide some builtin action to terminate process. It can only thrown panic which terminates just current thread. And if we want to test abnormal process termination we need to terminate the whole process.